samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/delivery/DeliveryActionListener.java

```diff
@@ -1352,7 +1352,7 @@ else if (answer !=null) {
     		   
     	}
     	if ( (item.getTypeId().equals(TypeIfc.EXTENDED_MATCHING_ITEMS)) || (item.getTypeId().equals(TypeIfc.MULTIPLE_CORRECT) )|| (item.getTypeId().equals(TypeIfc.MATCHING) )){
-    		if (mcmc_match_counter==correctAnswers){
+    		if (haswronganswer = false && mcmc_match_counter==correctAnswers){
```

Comment:
With the single equals sign you're setting the variable (always will return true).

If you want to test haswronganswer you can either say haswronganswer  == false or !haswronganswer